src/embezzler.ts

```diff
@@ -49,46 +51,131 @@ import { Macro, shouldRedigitize, withMacro } from "./combat";
 import { usingThumbRing } from "./dropsgear";
 import { crateStrategy, equipOrbIfDesired } from "./extrovermectin";
 import { baseMeat, globalOptions, ltbRun, WISH_VALUE } from "./lib";
-import { determineDraggableZoneAndEnsureAccess, draggableFight } from "./wanderer";
+import { familiarWaterBreathingEquipment, waterBreathingEquipment } from "./outfit";
+import { determineDraggableZoneAndEnsureAccess, DraggableFight } from "./wanderer";
 
-type EmbezzlerFightOptions = {
-  location?: Location;
-  macro?: Macro;
-};
+/**
+ * Configure the behavior of the fights in use in different parts of the fight engine
+ * @interface EmbezzlerFightConfigOptions
+ * @member {Requirement[]?} requirements maximizer requirements to use for this fight (defaults to empty)
+ * @member {draggableFight?} draggable if this fight can be pulled into another zone and what kind of draggable it is (defaults to undefined)
+ * @member {boolean?} canInitializeWandererCounters if this fight can be used to initialize wanderers (defaults to true)
+ * @member {boolean?} gregariousReplace if this is a "monster replacement" fight - pulls another monster from the CSV (defautls to false)
+ * @member {boolean?} wrongEncounterName if mafia does not update the lastEncounter properly when doing this fight (defaults to value of gregariousReplace)
+ */
+interface EmbezzlerFightConfigOptions {
+  requirements?: Requirement[];
+  draggable?: DraggableFight;
+  canInitializeWandererCounters?: boolean;
+  wrongEncounterName?: boolean;
+  gregariousReplace?: boolean;
+}
+
+class EmbezzlerFightRunOptions {
+  #macro: Macro;
+  #location?: Location;
+
+  constructor(macro: Macro, location?: Location) {
+    this.#macro = macro;
+    this.#location = location;
+  }
+
+  get macro(): Macro {
+    return this.#macro;
+  }
+
+  get location(): Location {
+    if (!this.#location) {
+      throw "Embezzler fight tried to access a location, but none was set";
+    } else {
+      return this.#location;
+    }
+  }
+}
 
 export class EmbezzlerFight {
+  name: string;
   available: () => boolean;
   potential: () => number;
-  run: (options: EmbezzlerFightOptions) => void;
+  execute: (options: EmbezzlerFightRunOptions) => void;
   requirements: Requirement[];
-  draggable: boolean;
-  name: string;
+  draggable?: DraggableFight;
+  canInitializeWandererCounters: boolean;
+  wrongEncounterName: boolean;
+  gregariousReplace: boolean;
+
   /**
    * This is the class that creates all the different ways to fight embezzlers
```

Comment:
so, EmbezzlerFights uses ~400 lines of embezzler.ts, and is the core of what makes garbo Different from a person's own random script. given the gigantic weight and importance of this class, i think this docstring needs some work. things i think it could benefit from:

- a formatting guide of what a properly written EmbezzlerFight actually looks like
- a clearer statement in the overarching summary that this is a Class That Runs a Fight; IMO, from a documentation standpoint, it is important to differentiate between "accounting" functions/classes and classes that are actually running turns for you
- a quick note as to why noob cave is the "default" zone. yeah, i know -- we all know why it is. someone reading garbo to try and build their own script may not, and if they were to ever change how noob cave operated, it would be good to have a description of what garbo's proposed viable conditions for a default zone is so we could quickly swap a new zone in.
- for the potential fights, we should maybe note that it is used by embezzlerCount to go into potions.ts to help stave off future circular references, since we don't docstring embezzlerCount (and probably shouldn't)